### PR TITLE
Fix: Time-of-check time-of-use filesystem race condition

### PR DESCRIPTION
**Pull request by brustolin**
Fixes [https://github.com/getsentry/sentry-cocoa/security/code-scanning/6](https://github.com/getsentry/sentry-cocoa/security/code-scanning/6)

To fix the TOCTOU race condition, we should avoid using `stat` and `open` separately. Instead, we can use the `open` function with the `O_RDONLY` flag to open the file directly and then use `fstat` to get the file status. This ensures that the file we are operating on is the same file we checked, as both operations are performed on the file descriptor.

_Suggested fixes powered by Copilot Autofix. Review carefully before merging._

### CODE REVIEW

File: Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c [hunk 1]
```diff
@@ -237,17 +237,18 @@ sentrycrashfu_readEntireFile(const char *const path, char **data, int *length, i
...
-        goto done;
-    }
 
     fd = open(path, O_RDONLY);
     if (fd < 0) {
         SENTRY_ASYNC_SAFE_LOG_ERROR("Could not open %s: %s", path, strerror(errno));
         goto done;
     }
 
+    if (fstat(fd, &st) < 0) {
```
Comment (brustolin) [inline Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c:247]:
```suggestion
    struct stat st;
    if (fstat(fd, &st) < 0) {
```

File: Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c [hunk 2]
```diff
@@ -237,17 +237,18 @@ sentrycrashfu_readEntireFile(const char *const path, char **data, int *length, i
     int bytesToRead = maxLength;
 
     struct stat st;
```
Comment (brustolin) [inline Sources/SentryCrash/Recording/Tools/SentryCrashFileUtils.c:239]:
```suggestion
```

**Review by philipphofmann — APPROVED** [review]
LGTM, but I think we should still ask somebody from the native team to have a look.

**Review by JoshuaMoelans — APPROVED** [review]
lgtm!

**Pull request merged**